src/components/IOUConfirmationList.js

```diff
@@ -328,84 +321,174 @@ class IOUConfirmationList extends Component {
     * @param {Object} option
     */
     toggleOption(option) {
-        // Return early if selected option is currently logged in user.
+        // Return early if selected option is currently logged-in user.
         if (option.login === this.props.session.email) {
             return;
         }
 
-        this.setState((prevState) => {
-            const newParticipants = _.map(prevState.participants, (participant) => {
-                if (participant.login === option.login) {
-                    return {...option, selected: !option.selected};
-                }
-                return participant;
-            });
-            return {participants: newParticipants};
-        });
+        this.setState(
+            (prevState) => {
+                const newParticipants = _.map(prevState.participants, (participant) => {
+                    if (participant.login === option.login) {
+                        return {...option, selected: !option.selected};
+                    }
+                    return participant;
+                });
+                const newSelectedParticipants = _.where(newParticipants, {selected: true});
+                const newSections = this.getSections(newParticipants);
+
+                return {
+                    sections: newSections,
+                    participants: newParticipants,
+                    selectedParticipants: newSelectedParticipants,
+                };
+            },
+            () => { this.allOptions = OptionsListUtils.flattenSections(this.state.sections); },
```

Comment:
**question:** Why is `allOptions` stored on the component instance instead of state?

Reply:
Because it doesn't directly affect rendering (only `scrollToIndex`), it doesn't need to be stored in state. By storing it on the component instance we can update it without triggering unnecessary re-renders.

Reply:
Does the arrow key manager need to know the length of the list has changed?

Reply:
https://github.com/Expensify/App/blob/2ad0ee3fb6da63a45871c38bb52c0e580d3cc082/src/components/IOUConfirmationList.js#L433-L438
https://github.com/Expensify/App/blob/2ad0ee3fb6da63a45871c38bb52c0e580d3cc082/src/components/ArrowKeyFocusManager.js#L70-L72